### ISSUE TITLE
blocking fifo open

### ISSUE THREAD

**Issue by hinxx** [cjj25/Yi-RTS3903N-RTSPServer#11]
I've compiled the `stream.c` and `rRTSPServer.cpp` using the https://github.com/cjj25/RTS3903N-rsdk-4.8.5-5281/blob/master/toolchains/rsdk-4.8.5-5281-EL-3.10-u0.9.33-m32fut-161202.tar.gz toolchain.

The result was placed to SD card and mounted on the device (not Yi). Does not seem to have the same uClibc libs as seen in this repo so I also placed `/toolchains/sdk/lib` shared libs on a separate folder on SD card.

Now I could execute the `stream` and `rRTSPServer` on the device, by first setting LD_LIBRARY_PATH.

Terminal 1:
```
/ # export LD_LIBRARY_PATH=/mnt/mmc/mmc1/sdk/lib:/mnt/mmc/mmc1/to_sd-3/lib
/ # /mnt/mmc/mmc1/to_sd-3/stream
current rts log mask 2
new rts log mask 1
current rts log level 254
new rts log level 254
rtstream : isp chn : 0
rtstream : h1 notify : 1, 0
rtstream : h264 chn : 10
rtstream : rts av run
set H264 to CVBR mode.
rtstream : Control thread started

```
Some extra debug was enabled/added to aid the analysis of what is happening.

Terminal 2:
```
/ # export LD_LIBRARY_PATH=/mnt/mmc/mmc1/sdk/lib:/mnt/mmc/mmc1/to_sd-3/lib
/ # /mnt/mmc/mmc1/to_sd-3/rRTSPServer 

"ch0_0.h264" stream, from the file "/tmp/h264_high_fifo"
Play this stream using the URL "rtsp://192.168.11.52/ch0_0.h264"

```

Terminal 3:
```
dummy@z800 ~ $ ffmpeg -v trace "rtsp://192.168.11.52/ch0_0.h264"
ffmpeg version 4.2.4-1ubuntu0.1 Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
  configuration: ... --enable-libx264 ...
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Splitting the commandline.
Reading option '-v' ... matched as option 'v' (set logging level) with argument 'trace'.
Reading option 'rtsp://192.168.11.52/ch0_0.h264' ... matched as output url.
Finished splitting the commandline.
Parsing a group of options: global .
Applying option v (set logging level) with argument trace.
Successfully parsed a group of options.
Parsing a group of options: output url rtsp://192.168.11.52/ch0_0.h264.
Successfully parsed a group of options.
Opening an output file: rtsp://192.168.11.52/ch0_0.h264.
Output #0, h264, to 'rtsp://192.168.11.52/ch0_0.h264':


Output file #0 does not contain any stream

```

In short the `ffmpeg` executed on a remote host does not seem to get any data?!

What I've observed is that in `stream` the call to `fd = fopen(FIFO_NAME_HIGH, "wb");` blocks and never returns. I would expect that once the `rRTSPServer` opens its side of the fifo for reading the `stream` would continue execution and start capturing frames. Fifo file `/tmp/h264_high_fifo` exists.

If I supply a normal file to write the frames to, i.e. `/tmp/stream-frames.h264` to `fopen()` the `stream` process happily writes frames to it and I can see a video on remote host afterwards. I've also confirmed that there is not an issue with using fifos on my device by running examples [here](https://www.geeksforgeeks.org/named-pipe-fifo-example-c-program/) with success.

**Comment by cjj25** [comment]
I'm a little late to review this one but great work! 

It looks like you've got it all figured out now but it's important to run the stream binary first and then the RTSPServer :)
